### PR TITLE
test(app):   Auth code flow flutter integration test IOS

### PR DESCRIPTION
**Pull request by talwinder50**
Android test will fail as need to figure out how to handle  cognito-mock.trustbloc.local:9229 redirect in the android

### CODE REVIEW

**Review by vkubiv — APPROVED** [review]

File: demo/app/integration_test/openid4ci_test.dart [hunk 1]
```diff
@@ -120,10 +120,77 @@ void main() async {
...
+        "scopes": ["openid","profile"],
+        "clientID": "oidc4vc_client",
+        "redirectURI": "http://127.0.0.1/callback"
+      };
+
+      var initializeResp = await walletSDKPlugin.initialize(issuanceURL, authCodeArgs);
+      var initializeRespEncoded = json.encode(initializeResp!);
+      Map<String, dynamic> initializeRespJson = json.decode(initializeRespEncoded);
+      var authorizationURLLink= initializeRespJson["authorizationURLLink"];
+    debugPrint("-------------------------");
```
Comment (DRK3) [inline demo/app/integration_test/openid4ci_test.dart:160]: Did you mean to leave in all these print statements?

File: demo/app/integration_test/openid4ci_test.dart [hunk 2]
```diff
@@ -120,10 +120,77 @@ void main() async {
...
     expect(matchedCreds, hasLength(equals(3)));
+
     expect(matchedCreds[0], equals(credentials[0]));
     expect(matchedCreds[1], equals(credentials[1]));
     expect(matchedCreds[2], equals(credentials[2]));
 
     await walletSDKPlugin.presentCredential();
   });
+
+  testWidgets('Testing openid4vc with a auth code flow', (tester) async {
```
Comment (DRK3) [inline demo/app/integration_test/openid4ci_test.dart:131]: Some typos here "a auth code flow"?

File: demo/app/integration_test/openid4ci_test.dart [hunk 3]
```diff
@@ -120,10 +120,77 @@ void main() async {
...
+    print("wallet didDocData : $didDocData");
+    var didDocEncoded = json.encode(didDocData!);
+    Map<String, dynamic> responseJson = json.decode(didDocEncoded);
+    var didContent= responseJson["did"];
+    print("wallet DID : $didContent");
+
+    const issuanceURL = String.fromEnvironment("INITIATE_ISSUANCE_URLS_AUTH_CODE_FLOW");
+    print("issuanceURLs in test: $issuanceURL");
+
+    const verificationURL = String.fromEnvironment("INITIATE_VERIFICATION_URLS");
```
Comment (DRK3) [inline demo/app/integration_test/openid4ci_test.dart:148]: Remove this line (`verificationURL` is unused)

File: demo/app/integration_test/openid4ci_test.dart [hunk 4]
```diff
@@ -120,10 +120,77 @@ void main() async {
...
+
+      var authCodeArgs =   {
+        "scopes": ["openid","profile"],
+        "clientID": "oidc4vc_client",
+        "redirectURI": "http://127.0.0.1/callback"
+      };
+
+      var initializeResp = await walletSDKPlugin.initialize(issuanceURL, authCodeArgs);
+      var initializeRespEncoded = json.encode(initializeResp!);
+      Map<String, dynamic> initializeRespJson = json.decode(initializeRespEncoded);
```
[the remark this answers is not in the document]
Comment (DRK3) [reply demo/app/integration_test/openid4ci_test.dart:158]: (this pattern should apply to the other areas where you use `json` in a name)

File: demo/app/integration_test/openid4ci_test.dart (hunk 1 above)
Comment (talwinder50) [reply demo/app/integration_test/openid4ci_test.dart:160]: yea couple of them for CI

**Review by rolsonquadras — APPROVED** [review]

**Pull request merged**